src/main/java/com/puppycrawl/tools/checkstyle/DetailAstImpl.java

```diff
@@ -20,22 +20,21 @@
 package com.puppycrawl.tools.checkstyle;
 
 import java.util.BitSet;
+import java.util.Collections;
+import java.util.List;
+
+import org.antlr.v4.runtime.Token;
 
-import antlr.CommonASTWithHiddenTokens;
-import antlr.Token;
-import antlr.collections.AST;
 import com.puppycrawl.tools.checkstyle.api.DetailAST;
 import com.puppycrawl.tools.checkstyle.utils.TokenUtil;
 
 /**
  * The implementation of {@link DetailAST}. This should only be directly used to
- * create custom AST nodes.
+ * create custom AST nodes and in 'JavaAstVisitor.java'.
  *
- * @noinspection FieldNotUsedInToString, SerializableHasSerializationMethods
+ * @noinspection FieldNotUsedInToString
  */
-public final class DetailAstImpl extends CommonASTWithHiddenTokens implements DetailAST {
-
-    private static final long serialVersionUID = -2580884815577559874L;
+public final class DetailAstImpl implements DetailAST {
```

Comment:
Many of the changes here were part of the ANTLR2 'AST' interface that we relied on https://www.antlr2.org/javadoc/antlr/collections/AST.html.